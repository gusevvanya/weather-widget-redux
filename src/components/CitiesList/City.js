import React from 'react';
import PropTypes from 'prop-types';

const City = (props) => {
	const onCityClick = (e) => {
		e.preventDefault();
		props.onCityClick(props.name);
	};

	const onDeleteCityClick = (e) => {
		e.stopPropagation();
		props.onDeleteCityClick(props.name);
	};

	return (
		<a 
			className={ props.className }
			onClick={onCityClick}
		>
			<span 
				className="close"
				onClick={onDeleteCityClick}
			>
				x
			</span>
			{props.name}
		</a>
	);   
};

City.propTypes = {
	name: PropTypes.string.isRequired,
	onCityClick: PropTypes.func.isRequired,
	onDeleteCityClick: PropTypes.func.isRequired,
	active: PropTypes.bool,
	className: PropTypes.string.isRequired,
};

export default City;
