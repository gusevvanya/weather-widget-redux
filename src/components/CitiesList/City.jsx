import React from 'react';
import PropTypes from 'prop-types';

const City = (props) => {
  const onCityClick = (e) => {
    e.preventDefault();
    props.onCityClick(props.name);
  };

  const onDeleteCityClick = (e) => {
    e.stopPropagation();
    props.onDeleteCityClick(props.name);
  };

  return (
    <a
      className={props.className}
      onClick={onCityClick}
    >
      <button
        className="close"
        onClick={onDeleteCityClick}
      >
        x
      </button>
      {props.name}
    </a>
  );
};

City.propTypes = {
  name: PropTypes.string.isRequired,
  onCityClick: PropTypes.func.isRequired,
  onDeleteCityClick: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired,
};

export default City;
