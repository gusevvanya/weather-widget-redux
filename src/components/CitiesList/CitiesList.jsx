import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import City from './City';

const CitiesList = ({
  cities,
  currentCity,
  onCityClick,
  onDeleteCityClick,
}) => {
  if (cities.length === 0) return null;

  const citiesList = cities.map((city) => {
    let active = false;

    if (currentCity === city) {
      active = true;
    }

    return (
      <City
        key={city}
        name={city}
        className={classNames('city', { active })}
        onCityClick={onCityClick}
        onDeleteCityClick={onDeleteCityClick}
      />
    );
  });

  return (
    <div className="cities">
      {citiesList}
    </div>
  );
};

CitiesList.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.string).isRequired,
  currentCity: PropTypes.string.isRequired,
  onCityClick: PropTypes.func.isRequired,
  onDeleteCityClick: PropTypes.func.isRequired,
};

export default CitiesList;
