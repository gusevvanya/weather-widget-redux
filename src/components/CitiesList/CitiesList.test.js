import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CitiesList from './CitiesList';
import City from './City';

Enzyme.configure({ adapter: new Adapter() });

describe('CitiesList', () => {
  let props;
  let citiesListMount;
  let citiesListShallow;

  beforeAll(() => {
    props = {
      cities: ['London'],
      currentCity: 'London',
      onCityClick: jest.fn(),
      onDeleteCityClick: jest.fn(),
    };
  });

  beforeEach(() => {
    jest.clearAllMocks();
    citiesListMount = Enzyme.mount(<CitiesList {...props} />);
    citiesListShallow = Enzyme.shallow(<CitiesList {...props} />);
  });

  it('should render', () => {
    expect(citiesListShallow.debug()).toMatchSnapshot();
  });

  it('render cities if the param "cities" recived', () => {
    expect(citiesListMount.find(City).length).toBe(1);
    citiesListMount.setProps({ cities: [] });
    expect(citiesListMount.find(City).length).toBe(0);
  });
});
