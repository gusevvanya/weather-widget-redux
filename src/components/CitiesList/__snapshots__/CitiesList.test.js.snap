// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList Should render 1`] = `
"<div className=\\"cities\\">
  <City name=\\"London\\" className=\\"city active\\" onCityClick={[Function]} onDeleteCityClick={[Function]} />
</div>"
`;
